### ISSUE TITLE
Database Diagram

### ISSUE THREAD

**Issue by kfish610** [Microsoft/sqlopsstudio#94]
Some form of Database Diagram, similar to SSMS, would make using this much easier and enjoyable.

**Comment by erickangMSFT** [comment]
[help wanted] Is anyone willing to build an extension for database diagram?

**Comment by BrentOzar** [comment]
When you say extension, would this be something that would be a part of this Github repo? If so, probably need to address the #102 licensing issue first before you accept code contributions.

**Comment by kevcunnane** [comment]
@BrentOzar no, extensions can develop outside this repo and use their own licensing. We plan to support extension installation in a future release.

**Comment by BrentOzar** [comment]
@kevcunnane WOOHOO! That's so awesome to hear. Thanks, sir.

**Comment by IshamMohamed** [comment]
Is extension supported yet?

**Comment by allanwei** [comment]
any update?

**Comment by JediSQL** [comment]
With database diagrams being removed from SSMS (SQL Server Management Studio) 18, someone should put a roaring fire under this.

**Comment by ScottSchwalm** [comment]
I agree with @JediSQL.  The priority on this should be moved towards the top of the pile with database diagrams removed from SSMS 18.

**Comment by JediSQL** [comment]
Suggestion 1: Per the agile paradigm, build the reverse engineering mode first; add forward engineering later.

**Comment by JediSQL** [comment]
Suggestion 2: It would be cool to have a feature that could, at least, read existing diagrams out of the sysdiagrams table. I've spent countless hours creating diagrams to support the developers that work with my databases.

**Comment by sanan-fataliyev** [comment]
+1

**Comment by mercuriete** [comment]
+1

**Comment by ibrowne** [comment]
+1

**Comment by waynebloss** [comment]
@JediSQL I just installed SSMS v18.4 and diagrams are there.

**Comment by mercuriete** [comment]
@waynebloss did you install azuredatastudio?
SSMS is another diferentes program.

**Comment by waynebloss** [comment]
@mercuriete I was responding to JediSQL who said "With database diagrams being removed from SSMS (SQL Server Management Studio) 18..."

I was simply correcting the information.

**Comment by JediSQL** [comment]
The original Microsoft documentation for removal of database diagrams from SSMS (SQL Server Management Studio) 18 said that it was **deprecated**. I think that someone got carried away and mistook deprecated for discontinued and removed it from SSMS 18.  So, Microsoft is  probably still planning on removing diagrams from a future version of SSMS.

**Comment by kburtram** [comment]
@JediSQL @waynebloss previously Database Diagrams was planned for removal from SSMS and was not included in 18.0.  It turns out customers actually like and use this feature 😄 so @Matteo-T added it back in a subsequent 18.# release.  I don't think there are currently any plans to remove diagrams from a future SSMS version as far as I'm aware.

**Comment by Matteo-T** [comment]
That is correct, @kburtram. Sadly, though, the Database Diagram is not fully functional at the moment. I've been trying to get it fix for quite some time now...

**Comment by tomaztk** [comment]
Please add Database Diagram to Azure data studio, similar to the one in SSMS. Thank you

**Comment by Adi-Sose** [comment]
+1

**Comment by mercuriete** [comment]
Please, it doesnt need to be fancy.

Something like graphviz -> SVG

In the end we only need tables and relations in a graph.

First iteration could be graphviz -> SVG only table names and relations.
something like: http://viz-js.com/
as azuredatastudio is open we can rely in another opensource projects.

**Comment by R0tenur** [comment]
I´ve built an extention for this. It doesn´t have that many features yet, but try it out if you are interested.

Here is the repo: https://github.com/R0tenur/visualization/

You can install from vsix downloaded here: https://github.com/R0tenur/visualization/releases/tag/v0.2.0

**Comment by fedelippo** [comment]
Please add an Entity Relationship Diagram tool to Azure data studio.

**Comment by bestickley** [comment]
Please add!

**Comment by AdrianoAE** [comment]
Would be very nice, even more if it allows IDEF1X

**Comment by mouradbka** [comment]
> vsix



> I´ve built an extention for this. It doesn´t have that many features yet, but try it out if you are interested.
> 
> Here is the repo: https://github.com/R0tenur/visualization/
> 
> You can install from vsix downloaded here: https://github.com/R0tenur/visualization/releases/

it worked great for me, thank you very much! I struggled a bit to install extension on azure data studio. What finally worked for me was to download the folder and then "file > install Extension from VSIX Package" and had to reload ADS and now it works seamlessly.

**Comment by AslanAmca** [comment]
Any progress?

**Comment by Az6bcn** [comment]
Any updates on this?

**Comment by huy312100** [comment]
Any update?

**Comment by JediSQL** [comment]
Looking @"Matteo-T commented on Dec 14, 2019" I get the impression that SSMS is where MSFT is putting their DB diagrams effort for the time being.

**Comment by Az6bcn** [comment]
Any updates on this?

**Comment by prajun7** [comment]
Still waiting

**Comment by diablo2947** [comment]
This would be a great help. Diagrams are really helpful in explaining the schema to people who are new on a project

**Comment by ederbond** [comment]
+1

**Comment by soumesh2002** [comment]
+1... It's been 2024 and there's no support for creating ER Diagram in Azure Data Studio officially. Hope in 2024 there will be support for creating ER Diagram.

**Comment by Az6bcn** [comment]
2 years after previopus comment... Any updates on this?

**Comment by princebillygk** [comment]
7 years after the issue was posted is there any update?

**Comment by ederbond** [comment]
Microsoft Soon™️.
Good at marketing, terrible at execution. I was wondering, why a trillion dollar company, can't hire a few developers to just do it in a few months?🤦🏻‍♂️

**Comment by steffes** [comment]
@chartdb 
https://github.com/chartdb/chartdb
https://app.chartdb.io/

**Comment by steffes** [comment]
@dbcodeio
https://github.com/dbcodeio/public

![Image](https://github.com/user-attachments/assets/6695cf52-8cff-4638-a552-8ca226bab715)

**Comment by ederbond** [comment]
@steffes do you know if there's the same extension, but for VS 2022?
Does it supports right clicking the table at the diagram to edit fields, indexes and etc and then when you save the changes on the ui it update both the diagram and the db structure as well?
